src/test/java/ymfc/commands/EditCommandTest.java

```diff
@@ -70,13 +71,14 @@ void testEditRecipe_success() throws IOException, InvalidArgumentException {
 
         editCommand.execute(emptyList, ingredientList, ui, storage);
         assertEquals(1, emptyList.getCounter());
-        assertEquals(edittedRecipe, emptyList.getRecipe(0));
+        assertEquals(editedRecipe.toString(), emptyList.getRecipe(0).toString());
     }
 
     @Test
     void testEditRecipe_fail() {
-        Recipe dummyRecipe = new Recipe("ramen", new ArrayList<Ingredient>(), new ArrayList<String>());
+        EditCommand dummyEditCommand = new EditCommand("ramen", null, null,
+                null, null, null);
         assertThrows(InvalidArgumentException.class,
-                () -> new EditCommand(dummyRecipe).execute(nonEmptyList, ingredientList, ui, storage));
+                () -> dummyEditCommand.execute(nonEmptyList, ingredientList, ui, storage));
```

Comment:
According to this test, an `InvalidArgumentException` should be raised when the `editCommand` only contains the `matchName` and nothing else, but I don't really see any lines among these changes handle that. Can you please enlighten me? 🤔

Reply:
Ah my bad, the test is tested for an non-existed recipe, but I've updated a follow-up suggestion, please do check!